gapis/resolve/command_tree.go

```diff
@@ -75,19 +75,30 @@ func (t *commandTree) index(indices []uint64) (api.SpanItem, api.SubCmdIdx) {
 	return group, subCmdRootID
 }
 
-func (t *commandTree) indices(id api.CmdID) []uint64 {
+func (t *commandTree) indices(idx []uint64) []uint64 {
 	out := []uint64{}
 	group := t.root
-	for {
-		i := group.IndexOf(id)
-		out = append(out, i)
-		switch item := group.Index(i).(type) {
-		case api.CmdIDGroup:
-			group = item
-		default:
-			return out
+
+	for _, id := range idx {
```

Comment:
jump_label:
for _, id := range idx {
    for {
          ....
          default:
            break jump_label
    }
}

Saves turning it into a coroutine.

Reply:
Removed the anonymous function in a way that won't displease Dijkstra